src/main/java/org/snpsift/lang/expression/FieldSub.java

```diff
@@ -100,7 +100,7 @@ public String getFieldString(VcfGenotype vcfGenotype) {
 		if (index < 0) {
 			FieldIterator.get().setMax(IteratorType.GENOTYPE_VAR, sub.length - 1);
 			FieldIterator.get().setType(index);
-			idx = FieldIterator.get().get(IteratorType.VAR);
+			idx = FieldIterator.get().get(IteratorType.GENOTYPE_VAR);
```

Comment:
Not sure about this change, but seems like this should be `GENOTYPE_VAR`.
Otherwise this need to be changed to
```
idx = FieldIterator.get().getVar(name)
```
Would recommend adding a test case that covers this use case.